Finder.arrayOf(...) always returns Object[] if an optional view is null

If you bind an array of views and one is null/optional, `Finder.arrayOf(…)` returns an instance of `Object[]`, which results in a `ClassCastException` at runtime.

This worked fine before 7.0.0 when we used `@InjectViews`, I'm not sure what changed under the hood. It might be the case you shouldn't try to have nullable view in an array anyway, but just wanted to let you know in case this wasn't intentional.

### Example

```java
@Nullable
@Bind({R.id.textview0, R.id.textview1, R.id.textview2})
TextView[] textViews;
```

Results in this crash at runtime:

```
Caused by: java.lang.ClassCastException: java.lang.Object[] cannot be cast to android.widget.TextView[]
            at io.sweers.test.MainActivity$$ViewBinder.bind(MainActivity$$ViewBinder.java:11)
            at io.sweers.test.MainActivity$$ViewBinder.bind(MainActivity$$ViewBinder.java:8)
            at butterknife.ButterKnife.bind(ButterKnife.java:328)
            at butterknife.ButterKnife.bind(ButterKnife.java:246)
            at io.sweers.test.MainActivity.onCreate(MainActivity.java:23)
```

Full example here: https://github.com/hzsweers/Butterknife-test

--- a/butterknife/src/main/java/butterknife/ButterKnife.java
+++ b/butterknife/src/main/java/butterknife/ButterKnife.java
@@ -9,6 +9,7 @@ import android.util.Log;
 import android.util.Property;
 import android.view.View;
 import butterknife.internal.ButterKnifeProcessor;
+import java.util.Arrays;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
@@ -114,24 +115,14 @@ public final class ButterKnife {
     };
 
     private static <T> T[] filterNull(T[] views) {
-      int newSize = views.length;
-      for (T view : views) {
-        if (view == null) {
-          newSize -= 1;
-        }
-      }
-      if (newSize == views.length) {
-        return views;
-      }
-      //noinspection unchecked
-      T[] newViews = (T[]) new Object[newSize];
-      int nextIndex = 0;
-      for (T view : views) {
+      int end = 0;
+      for (int i = 0; i < views.length; i++) {
+        T view = views[i];
         if (view != null) {
-          newViews[nextIndex++] = view;
+          views[end++] = view;
         }
       }
-      return newViews;
+      return Arrays.copyOfRange(views, 0, end);
     }
 
     public static <T> T[] arrayOf(T... views) {

